src/main/java/com/rainbowrave/RainbowRaveMouseTrailOverlay.java

```diff
@@ -0,0 +1,162 @@
+    }
+
+    // Function the get the rainbow color for a particular point based on the size of the trail
+    private Color getRainbowColor(int minHue, int trailSize, int position) {
+        double currentPercent = (double) (position - minHue) / (trailSize - minHue);
+        float hue = (float) (currentPercent * (trailSize - minHue) + minHue);
+        return Color.getHSBColor(hue / 100, 1.0f, 1.0f);
+    }
+
+    // Helper method to handle choosing the correct coloring function
+    private Color getColor(int size, int position) {
+        if(rainbowRaveConfig.whichMouseTrailStyle() == RainbowRaveConfig.MouseTrailStyle.PARTYMODE) {
+            return getRainbowColor(0, size, position);
+        } else if (rainbowRaveConfig.whichMouseTrailStyle() == RainbowRaveConfig.MouseTrailStyle.ENABLED) {
+            return getRainbowColor(0, size, position);
+        } else if(rainbowRaveConfig.whichMouseTrailStyle() == RainbowRaveConfig.MouseTrailStyle.SYNCED) {
+            return rainbowRavePlugin.getColor(0);
+        } else {
+            // Fall back to standard coloring
+            return getRainbowColor(0, size, position);
+        }
+    }
+
+    @Override
+    public Dimension render(Graphics2D graphics)
+    {
+        // Disable overlay if mouse trail style set to None
+        if (rainbowRaveConfig.whichMouseTrailStyle() == RainbowRaveConfig.MouseTrailStyle.NONE) {
+            return null;
+        }
+        // Set Partymode variables
+        boolean isPartyMode = rainbowRaveConfig.whichMouseTrailStyle() == RainbowRaveConfig.MouseTrailStyle.PARTYMODE;
+        // Get ArrayList of Curves
+        ArrayList<Curve> trail = new ArrayList<>(plugin.getTrail());
+        // Points to track where to render a line between Curves
+        Point midBefore = null;
+        Point midAfter = null;
+
+        // Set trail size, stroke, and antialiasing
+        graphics.setStroke(new BasicStroke(2, BasicStroke.CAP_BUTT, BasicStroke.JOIN_ROUND));
+        graphics.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON);
+        // Fallback default color
+        graphics.setColor(Color.BLUE);
+        // Loop through Curves of trail
+        for(int i = 0; i < trail.size(); i++) {
+            // Get Points from Curve
+            ArrayList<Point> points = trail.get(i).getCurve();
+            // Loop through points
+            for(int j = 0; j < points.size(); j++ ) {
+
+                // Initialize points used to calculate parts of trail with null
+                Point before = null;
+                Point after = null;
+                Point previous = null;
+                Point current = null;
+                // Logic for setting points depending on where in the loops we are
+                if(i != 0 && j == 0) {
+                    // Get previous list of points from the previous Curve
+                    ArrayList<Point> previousPoints = trail.get(i - 1).getCurve();
```

Comment:
List not ArrayList.

Reply:
fyi using ArrayList for variable types is a java faux pas. It probably causes issues if you use something like Collections.emptyList().